src/kpro_connection.erl

```diff
@@ -295,38 +295,39 @@ get_tcp_mod(_SslOpts = true)  -> ssl;
 get_tcp_mod(_SslOpts = [_|_]) -> ssl;
 get_tcp_mod(_)                -> gen_tcp.
 
-%% If SslOpts contains {verify, verify_peer}, we insert
-%% {server_name_indication, Host}. This is necessary as of OTP 20, to
-%% ensure that peer verification is done against the correct host name
-%% (otherwise the IP will be used, which is almost certainly
-%% incorrect).
-insert_server_name_indication(SslOpts, Host) ->
-  VerifyOpt = proplists:get_value(verify, SslOpts),
-  insert_server_name_indication(VerifyOpt, SslOpts, Host).
-
-insert_server_name_indication(verify_peer, SslOpts, Host) ->
-  case proplists:get_value(server_name_indication, SslOpts) of
-    undefined ->
-      %% insert {server_name_indication, Host} if not already present
-      [{server_name_indication, ensure_string(Host)} | SslOpts];
-    _ ->
-      SslOpts
-  end;
-
-insert_server_name_indication(_, SslOpts, _) ->
-  SslOpts.
+%% insert {server_name_indication, Host} if not already present
+%% Some special values are allowed:
+%% * auto: use the host as SNI
+%% * none: do not use SNI
+%% * undefined: same as auto (for backward compatibility)
+insert_server_name_indication(SslOpts0, Host) ->
+  SNI = ensure_string(proplists:get_value(server_name_indication, SslOpts0)),
```

Comment:
Tiny nit: maybe we could ensure it is a string only if it's not one of the expected atoms (undefined, none, auto) to distinguish them from input?

Reply:
allowing in string format `"auto"` and `"none"` will make it easier for configs parsed from HOCON or JSON so there is no need to convert to atom.